comictalker/comictalkerapi.py

```diff
@@ -16,23 +16,50 @@
 from __future__ import annotations
 
 import logging
+import sys
+from typing import NamedTuple
+
+if sys.version_info < (3, 10):
+    from importlib_metadata import EntryPoint, distributions
+else:
+    from importlib.metadata import EntryPoint, distributions
 
-import comictalker.talkers.comicvine
 from comictalker.talkerbase import ComicTalker, TalkerError
 
 logger = logging.getLogger(__name__)
 
 
-def get_comic_talker(source_name: str) -> type[ComicTalker]:
-    """Retrieve the available sources modules"""
-    sources = get_talkers()
-    if source_name not in sources:
-        raise TalkerError(source=source_name, code=4, desc="The talker does not exist")
+class Plugin(NamedTuple):
+    package: str
+    version: str
+    entry_point: EntryPoint
+
+
+def get_comic_talker(talker_name: str) -> type[ComicTalker]:
+    """Returns the requested talker module"""
+    talkers = get_talkers()
+    if talker_name not in talkers:
+        raise TalkerError(source=talker_name, code=4, desc="The talker does not exist")
+
+    return talkers[talker_name].entry_point.load()
+
+
+def get_talkers() -> dict[str, Plugin]:
+    """Returns all comic talker plugins (internal and external)"""
+    talkers: dict[str, Plugin] = {}
+    for dist in distributions():
+        eps = dist.entry_points
+
+        # perf: skip parsing `.metadata` (slow) if no entry points match
+        if not any(ep.group in ("comictagger_talkers") for ep in eps):
+            continue
 
-    talker = sources[source_name]
-    return talker
+        # assigned to prevent continual reparsing
+        meta = dist.metadata
 
+        for ep in eps:
+            # Only want comic talkers
+            if ep.group == "comictagger_talkers":
+                talkers[ep.name] = Plugin(ep.name, meta["version"], ep)
```

Comment:
This should load the entry point we don't have a reason not to and it needs a try block to make sure that the entry point loads successfully
```suggestion
        for ep in eps:
            # Only want comic talkers
            if ep.group == "comictagger_talkers":
                try:
                    talkers[ep.name] = ep.load()
                except Exception:
                    logger.exception("Failed to load talker: %s", ep.name)
```

Reply:
That is basically what I had originally but I wanted to get the version number in and followed flake8. I don't think the cache is needed if, for now, there is only one talker and when the option to switch talkers will be in another PR. In that PR, I don't think `get_comic_talker` will be needed. Do you want me to add it in now anyway?

Reply:
If you don't want to cache it, remove the function for retrieving an individual Talker. It's a glorified dict lookup.

Reply:
Removed `cache` and `get_comic_talker`. `main` calls `get_talkers` and stores it in `self.talker_plugins` now on `init`.